lib/spack/spack/compilers/clang.py

```diff
@@ -107,3 +109,48 @@ def default_version(cls, comp):
             cpr._version_cache[comp] = ver
 
         return cpr._version_cache[comp]
+
+    def setup_custom_environment(self, env):
+        super(Clang, self).setup_custom_environment(env)
+
+        if not self.is_apple:
```

Comment:
i think you also need to check if compiler is `clang-apple` as someone may be using non-aple clang on macOS (i don't).